packages/edit-post/src/store/effects.js

```diff
@@ -43,78 +122,13 @@ const effects = {
 			}
 		} );
 
-		let wasSavingPost = select( 'core/editor' ).isSavingPost();
-		let wasAutosavingPost = select( 'core/editor' ).isAutosavingPost();
-		let wasPreviewingPost = select( 'core/editor' ).isPreviewingPost();
-		// Save metaboxes when performing a full save on the post.
-		subscribe( () => {
-			const isSavingPost = select( 'core/editor' ).isSavingPost();
-			const isAutosavingPost = select( 'core/editor' ).isAutosavingPost();
-			const isPreviewingPost = select( 'core/editor' ).isPreviewingPost();
-			const hasActiveMetaBoxes = select( 'core/edit-post' ).hasMetaBoxes();
-
-			// Save metaboxes on save completion, except for autosaves that are not a post preview.
-			const shouldTriggerMetaboxesSave = (
-				hasActiveMetaBoxes && (
-					( wasSavingPost && ! isSavingPost && ! wasAutosavingPost ) ||
-					( wasAutosavingPost && wasPreviewingPost && ! isPreviewingPost )
-				)
+		if ( hasMetaBoxes( store.getState() ) ) {
+			addFilter(
```

Comment:
Does `addFilter` prevent multiple filters being registered or could it be worth a call to `removeFilter` first, in case this `SET_META_BOXES_PER_LOCATIONS` action is called more than once?

Reply:
Cursory glance at the code and it looks like I should probably add a call to `removeFilter` to be safe.

Reply:
yes, otherwise you will add multiple callbacks